### ISSUE TITLE
FIM v2.0: Error message when deleting/renaming file in whodata in Windows

### ISSUE THREAD

**Issue by jesusjimsa** [wazuh/wazuh#4654]
| Wazuh version | Component        | Install type          | Install method         | Platform |
|----------------|-----------------|-----------------|--------------------|----------|
| 3.12                   | ossec-syscheck | Manager/Agent  | Packages/Sources | Windows |

## Bug description

When deleting/renaming a file in a folder monitored with `whodata`, an error from `GetNamedSecurityInfo()` appears in the log.  

Despite the error message, all the alerts are received properly.

## Log

```Log
2020/02/24 15:13:34 ossec-agent[6508] win_whodata.c:770 at whodata_callback(): DEBUG: (6231): The 'c:\whodatanew' directory has been scanned after detecting event of new files.
2020/02/24 15:13:34 ossec-agent[6508] win_whodata.c:1510 at w_update_sacl(): ERROR: (6686): GetNamedSecurityInfo() failed. Error '2'
2020/02/24 15:13:34 ossec-agent[6508] create_db.c:100 at fim_checker(): DEBUG: (6288): Could not refresh the SACL of 'c:\whodatanew\new text document.txt'. Its event will not be reported.
2020/02/24 15:13:34 ossec-agent[6508] fim_db.c:435 at fim_db_check_transaction(): DEBUG: Database transaction completed.
```

**Comment by jotacarma90** [comment]
The problem when deleting a file was that the GetNamedSecurityInfo function was trying to get the information from a path that no longer exist. 
The function giving that path was w_update_sacl(path). It is necessary to change the order in which this function is called, so first it has been checked if the path exists.

**Comment by jotacarma90** [comment]
This issue will be solved together with others in a major one with anothers fixes.

**Comment by bah07** [comment]
Fixed in [#4680](https://github.com/wazuh/wazuh/pull/4680)
